src/script/interpreter.cpp

```diff
@@ -485,6 +484,7 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                         if (opcode == OP_NOTIF)
                             fValue = !fValue;
                         popstack(stack);
+                        fExec = fValue;
```

Comment:
in `OP_IF`, if `fExec` is true, the original `vfExec` must be by definition all true. Since `fValue` is added to the end of `vfExec`, the new `fExec` value must be same as `fValue`

If `fExec` is false, a false will be added to `vfExec`, and by definition `fExec` will remain false, so we don't need to do anything